### PR TITLE
Close #11561: Use ScreenCoordsXY in gfx_draw_string_centred_wrapped()

### LINKED ISSUE #11561: Use ScreenCoordsXY on gfx_draw_string_centred_wrapped
Create an overload of `gfx_draw_string_centred_wrapped` on `\src\openrct2\drawing\Text.cpp` with the same name, but that takes a `const ScreenCoordsXY&` object instead of an `int32_t x` and and an `int32_t y` and replace the calls to the `{x ,y}` pair by the one using `ScreenCoordsXY`.

Don't try to do all calls at once, unless it doesn't occur a lot, as this might lead to a huge PR. When we stop using the old function, we can drop it entirely in favour of the `ScreenCoordsXY` alternative.

### PR DESCRIPTION
**Pull request by frutiemax**
It aims to use the gfx_draw_string_centred_wrapped to use ScreenCoordsXY.

### CODE REVIEW

**Comment by tupaschoal** [comment]
(Mentioning that this is part of #11561 so it appears there)

File: src/openrct2-ui/windows/About.cpp [hunk 1]
```diff
@@ -202,55 +202,54 @@ static void window_about_openrct2_common_paint(rct_window* w, rct_drawpixelinfo*
...
     window_draw_widgets(w, dpi);
 
-    int32_t x1, x2, y;
-
-    x1 = w->windowPos.x + (&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->left + 45;
-    x2 = w->windowPos.x + (&w->widgets[WIDX_TAB_ABOUT_RCT2])->left + 45;
-    y = w->windowPos.y + (((&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->top + (&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->bottom) / 2)
-        - 3;
+    int32_t y =
+        w->windowPos.y + (((&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->top + (&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->bottom) / 2) - 3;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/About.cpp:206]: This is pretty horrible to read. Could you create a temporary ABOUT_OPENRCT2 widget something like:

```
auto aboutWidget = &w->widgets[WIDX_TAB_ABOUT_OPENRCT2];
y = w->windowPos.y + (aboutWidget.top + aboutWidget.bottom) / 2 - 3;
ScreenCoordsXY coord1(w->windowPos.x + aboutWidget.left + 45, y);
```

**Comment by duncanspumpkin** [comment]
Can you rebase and also check your formatting.

**Comment by janisozaur** [comment]
Either commits should have meaningful subjects and contents or this needs to be squashed with a proper commit message.

File: src/openrct2/drawing/Drawing.h [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include "../common.h"
 #include "../interface/Colour.h"
 #include "../interface/ZoomLevel.hpp"
+#include "../world/Location.hpp"
```
Comment (tupaschoal) [inline src/openrct2/drawing/Drawing.h:16]: Can you remove this and forward declare `ScreenCoordsXY` instead?

File: src/openrct2/drawing/Text.cpp [hunk 3]
```diff
@@ -247,3 +247,28 @@ int32_t gfx_draw_string_centred_wrapped(
 
     return layout.GetHeight();
```
Comment (tupaschoal) [inline src/openrct2/drawing/Text.cpp]: Please make the non `ScreenCoordsXY` function for `gfx_draw_string_centred_wrapped` call the new one, so we don't duplicate code.

File: src/openrct2-ui/windows/About.cpp [hunk 4]
```diff
@@ -218,37 +218,39 @@ static void window_about_openrct2_paint(rct_window* w, rct_drawpixelinfo* dpi)
...
 
-    int32_t x, y, width;
+    int32_t width;
     rct_size16 logoSize;
 
     int32_t lineHeight = font_get_line_height(gCurrentFontSpriteBase);
 
-    x = w->windowPos.x + (w->width / 2);
-    y = w->windowPos.y + w->widgets[WIDX_PAGE_BACKGROUND].top + lineHeight;
+    ScreenCoordsXY coord(w->windowPos.x + (w->width / 2), w->windowPos.y + w->widgets[WIDX_PAGE_BACKGROUND].top + lineHeight);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/About.cpp:228]:
```suggestion
    ScreenCoordsXY aboutCoords (w->windowPos.x + (w->width / 2), w->windowPos.y + w->widgets[WIDX_PAGE_BACKGROUND].top + lineHeight);
```
Give a more meaningful name to the variable. This is just a suggestion, you might think of others.

This applies to other places in the diff too.

File: src/openrct2-ui/windows/TrackDesignManage.cpp [hunk 5]
```diff
@@ -292,7 +292,7 @@ static void window_track_delete_prompt_paint(rct_window* w, rct_drawpixelinfo* d
     window_draw_widgets(w, dpi);
 
     gfx_draw_string_centred_wrapped(
-        dpi, &_trackDesignFileReference->name, w->windowPos.x + 125, w->windowPos.y + 28, 246,
+        dpi, &_trackDesignFileReference->name, ScreenCoordsXY(w->windowPos.x + 125, w->windowPos.y + 28), 246,
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignManage.cpp:295]: You can do just
```suggestion
        dpi, &_trackDesignFileReference->name, { w->windowPos.x + 125, w->windowPos.y + 28 }, 246,
```

File: src/openrct2-ui/windows/About.cpp [hunk 6]
```diff
@@ -202,10 +202,12 @@ static void window_about_openrct2_common_paint(rct_window* w, rct_drawpixelinfo*
...
-    int32_t y = w->windowPos.y
-        + (((&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->top + (&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->bottom) / 2) - 3;
-    ScreenCoordsXY coord1(w->windowPos.x + (&w->widgets[WIDX_TAB_ABOUT_OPENRCT2])->left + 45, y);
-    ScreenCoordsXY coord2(w->windowPos.x + (&w->widgets[WIDX_TAB_ABOUT_RCT2])->left + 45, y);
+    const auto& aboutOpenRCT2 = w->widgets[WIDX_TAB_ABOUT_OPENRCT2];
+    const auto& aboutRCT2 = w->widgets[WIDX_TAB_ABOUT_RCT2];
+
+    int32_t y = w->windowPos.y + ((aboutOpenRCT2.top + aboutOpenRCT2.bottom) / 2) - 3;
+    ScreenCoordsXY coord1(w->windowPos.x + aboutOpenRCT2.left + 45, y);
+    ScreenCoordsXY coord2(w->windowPos.x + aboutRCT2.left + 45, y);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/About.cpp:210]: meaningful names

File: src/openrct2-ui/windows/Ride.cpp [hunk 7]
```diff
@@ -6192,10 +6194,9 @@ static void window_ride_graphs_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi
     if (measurement == nullptr)
     {
         // No measurement message
-        int32_t x = (widget->right - widget->left) / 2;
-        int32_t y = (widget->bottom - widget->top) / 2 - 5;
+        ScreenCoordsXY coord((widget->right - widget->left) / 2, (widget->bottom - widget->top) / 2 - 5);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Ride.cpp:6197]: coord**s**. There are two of them, after all.

File: src/openrct2-ui/windows/TitleScenarioSelect.cpp [hunk 8]
```diff
@@ -462,7 +459,10 @@ static void window_scenarioselect_paint(rct_window* w, rct_drawpixelinfo* dpi)
         { // old-style
             ft.Add<rct_string_id>(ScenarioCategoryStringIds[i]);
         }
-        gfx_draw_string_centred_wrapped(dpi, gCommonFormatArgs, x, y, 87, format, COLOUR_AQUAMARINE);
+
+        ScreenCoordsXY stringCoords(
+            (widget->left + widget->right) / 2 + w->windowPos.x, (widget->top + widget->bottom) / 2 + w->windowPos.y - 3);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/TitleScenarioSelect.cpp:464]: If you put a return after the comma, does clang-format accept that?

Comment (frutiemax) [reply src/openrct2-ui/windows/TitleScenarioSelect.cpp:464]: No it doesn't, it reformats this way.

**Comment by tupaschoal** [comment]
Thanks @frutiemax !

**Pull request merged**